src/network/networkctl.c

```diff
@@ -2919,6 +2927,250 @@ static int verb_reconfigure(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int get_arg_drop_in(char **ret) {
+        _cleanup_free_ char *drop_in = NULL;
+
+        assert(ret);
+
+        if (isempty(arg_drop_in))
+                return 0;
+        else if (!endswith(arg_drop_in, ".conf")) {
+                drop_in = strjoin(arg_drop_in, ".conf");
+        } else
+                drop_in = strdup(arg_drop_in);
+        if (!drop_in)
+                return log_oom();
+
+        if (!filename_is_valid(drop_in))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid drop-in file name '%s'.", drop_in);
+
+        *ret = TAKE_PTR(drop_in);
+        return 1;
+}
+
+static int get_config_files_by_name(const char *name, char **ret_path, char ***ret_dropins) {
+        _cleanup_strv_free_ char **dropins = NULL, **dropin_dirs = NULL;
+        _cleanup_free_ char *path = NULL;
+        int r;
+
+        assert(name);
+        assert(ret_path);
+
+        if (!ENDSWITH_SET(name, ".network", ".netdev", ".link"))
+                return -EINVAL;
+
+        STRV_FOREACH(i, NETWORK_DIRS) {
+                _cleanup_free_ char *p = NULL;
+
+                p = path_join(*i, name);
+                if (!p)
+                        return -ENOMEM;
+
+                r = strv_extendf(&dropin_dirs, "%s.d", p);
+                if (r < 0)
+                        return r;
+
+                if (!path && access(p, F_OK) >= 0)
+                        path = TAKE_PTR(p);
+        }
+
+        if (!path)
+                return -ENOENT;
+
+        if (!ret_dropins) {
+                *ret_path = TAKE_PTR(path);
+                return 0;
+        }
+
+        r = conf_files_list_strv(&dropins, ".conf", NULL, 0, (const char* const*) dropin_dirs);
+        if (r < 0)
+                return r;
+
+        *ret_path = TAKE_PTR(path);
+        *ret_dropins = TAKE_PTR(dropins);
+
+        return 0;
+}
+
+static int get_dropin_by_name(
+                const char *name,
+                char * const *dropins,
+                char **ret) {
+
+        assert(name);
+        assert(dropins);
+        assert(ret);
+
+        STRV_FOREACH(i, dropins)
+                if (path_equal_filename(*i, name)) {
+                        _cleanup_free_ char *d = NULL;
+
+                        d = strdup(*i);
+                        if (!d)
+                                return -ENOMEM;
+
+                        *ret = TAKE_PTR(d);
+                        return 1;
+                }
+
+        return 0;
+}
+
+static int add_config_to_edit(
+                EditFileContext *context,
+                const char *main,
```

Comment:
Please do not use the name `main`, as we have `main()`, and we usually avoid such conflicting name for safety.